Source/WebKit/UIProcess/Extensions/gtk/WebExtensionGtk.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2024 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WebExtension.h"
+
+#include "Logging.h"
+
+namespace WebKit {
+
+static String const generatedBackgroundPageFilename = "_generated_background_page.html"_s;
+static String const generatedBackgroundServiceWorkerFilename = "_generated_service_worker.js"_s;
+
+std::span<const uint8_t> WebExtension::resourceDataForPath(String path, CacheResult cacheResult, SuppressNotFoundErrors suppressErrors)
```

Comment:
At a glance places that use this have a `URL` for path, keep that.

Then below use functions like `URL.protocolIsData()`.

Reply:
i can't find a place that would normally use a URL on this function

Reply:
It does here: https://github.com/ItsJamie9494/WebKit/pull/1/files#diff-3d7aff3ea45a3d96af4bf4f21e1621be2cfe0193d89155ca68d614b5a0446bb4R109

But that does seem to be an exception, so this is fine.